File: dsa-practice/sorting/python/Bubble/basic.py
def bubble_sort(arr):
    n = len(arr)
    # Outer loop for passes through the array
    for i in range(n):
        # Inner loop for comparisons and swaps in each pass
        # The last 'i' elements are already in place, so we don't need to check them
        for j in range(0, n - i - 1):
            # Compare adjacent elements
            if arr[j] > arr[j + 1]:
                # Swap if they are in the wrong order
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
    return arr

# Example usage:
my_list = [64, 34, 25, 12, 22, 11, 90]
sorted_list = bubble_sort(my_list)
print(f"Sorted array: {sorted_list}")

File: dsa-practice/sorting/java/Bubble/basic.java

public class basic {

    static void bubblesort(int arr[], int n) {
        int k , k2 , temp ;
        boolean swapcheck = false;
        

        
        for ( k = 0; k < n - 1; k++) {
            swapcheck = false ;
            for ( k2 = 0; k2 < n - k - 1; k2++) {
                //swap
                if (arr[k2]> arr[k]) {
                    temp = arr[k2];
                    arr[k2] = arr[k];
                    arr[k] = temp;
                    swapcheck = true;
                }
                
            }
            
            
        }
        if (swapcheck == false) {
                System.out.println("No swapping");
                
            }

        
    }

    static void printarr(int arr[], int n) {
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i]+" ");
            
        }System.out.println("");

    }

    public static void main(String args[]) {
        int arr[] = {5, 1, 4, 2, 8};
        int n = arr.length;
        bubblesort(arr, n);
        System.out.print("Sorted Arr = ");
        printarr(arr, n);

    }
}